src/compiler/checker.rs

```diff
@@ -2548,6 +2830,56 @@ impl NodeBuilder {
         chain
     }
 
+    fn symbol_to_type_node(
+        &self,
+        type_checker: &TypeChecker,
+        symbol: Rc<Symbol>,
+        context: &NodeBuilderContext,
+        meaning: SymbolFlags,
+    ) -> TypeNode {
+        let chain = self.lookup_symbol_chain(symbol, context, Some(meaning));
+
+        let chain_index = chain.len() - 1;
+        let entity_name =
+            self.create_access_from_symbol_chain(type_checker, context, chain, chain_index, 0);
+        if false {
+            unimplemented!()
+        } else {
+            factory
+                .create_type_reference_node(&self.synthetic_factory, entity_name)
+                .into()
+        }
+    }
+
+    fn create_access_from_symbol_chain(
+        &self,
+        type_checker: &TypeChecker,
+        context: &NodeBuilderContext,
+        chain: Vec<Rc<Symbol>>,
+        index: usize,
+        stopper: usize,
+    ) -> Rc<Node> {
+        let symbol = chain[index].clone();
+
+        let mut symbol_name: Option<String>;
+        if index == 0 {
+            symbol_name =
+                Some(type_checker.get_name_of_symbol_as_written(symbol.clone(), Some(context)));
+        } else {
+            unimplemented!()
+        }
+        if symbol_name.is_none() {
```

Comment:
This seems impossible to be true, but is that just because the `None` would come from the thing that's currently `unimplemented`?